flare/gp.py

```diff
@@ -682,33 +625,29 @@ def as_dict(self):
 
         return out_dict
 
+    def sync_data(self):
```

Comment:
Maybe we should move the setting of the NP training labels to this function, so that it's only a matter of adding to training data and training labels?

Reply:
Updating the 'all labels' attribute could also happen here.